### ISSUE TITLE
Private Endpoint trying to recreate endpoint over and over since v3.44.0 with redis

### ISSUE THREAD

**Issue by maonat** [hashicorp/terraform-provider-azurerm#20805]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.6

### AzureRM Provider Version

3.44.0

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_virtual_network" "example" {
  name                = "example-network"
  address_space       = ["10.0.0.0/16"]
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
}

resource "azurerm_subnet" "service" {
  name                 = "service"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.1.0/24"]

  enforce_private_link_service_network_policies = true
}

resource "azurerm_subnet" "endpoint" {
  name                 = "endpoint"
  resource_group_name  = azurerm_resource_group.example.name
  virtual_network_name = azurerm_virtual_network.example.name
  address_prefixes     = ["10.0.2.0/24"]

  enforce_private_link_endpoint_network_policies = true
}

resource "azurerm_redis_cache" "example" {
  name                = "example-cache"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  capacity            = 2
  family              = "C"
  sku_name            = "Standard"
  enable_non_ssl_port = false
  minimum_tls_version = "1.2"

  redis_configuration {
  }
}

resource "azurerm_private_endpoint" "example" {
  name                = "example-endpoint"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  subnet_id           = azurerm_subnet.endpoint.id

  private_service_connection {
    name                           = azurerm_redis_cache.example.name
    private_connection_resource_id = azurerm_redis_cache.example.id
    is_manual_connection           = false
    subresource_names              = ["redisCache"]
  }
}
```

### Debug Output/Panic Output

```shell
N/A
```

### Expected Behaviour

After applying same tf file a second time, I expect the private ip to not trigger a second time the creation of a new private endpoint.

### Actual Behaviour

Once the private endpoint is created, terraform show again that the ID of the redis resource is changed because a capital letter in the ID of the resource. The capital letter is in the word redis (located right after /Microsoft.Cache/)

In the state file the ID provided by the redis cache server is correct, unfortunately looks like that the endpoint is thinking that the resource is not the same which trigger the destruction of the already existing endpoint and its recreation.

What triggers the changes is private_connection_resource_id.

Example below:

![CleanShot 2023-03-06 at 14 19 44](https://user-images.githubusercontent.com/56087756/223126850-8e3f60b6-1227-49d0-bc6c-65d7f7f302b8.png)

### Steps to Reproduce

1. `terraform apply`
2. `terraform apply`

### Important Factoids

By using 3.43.0 you won't incur in the issue.

**Comment by maonat** [comment]
Why the BUG tag has been removed? 👎

**Comment by jbinko** [comment]
I was not able to reproduce it in the version 3.46. I haven't tried previous versions

**Comment by sinbai** [comment]
@maonat thanks for opening this issue. It seems to be the same as issue #20413 and has been fixed in terraform provider azurerm release [v3.44.1](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.44.1/CHANGELOG.md) . Could you upgrade the tf provider to see if it is resolved?

**Comment by manicminer** [comment]
Since it looks like a fix for this was already released, I'm going to go ahead and close this issue for now. If this is not correct, please feel free to comment further and we'll reopen this. Thanks!
